src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -275,3 +351,233 @@ async function fetchAndCachePossibleDomNode(
 
   testEventDomNodeCache.cacheValue(domNodeDetails, domNodeDetails.point);
 }
+
+async function fetchPlaywrightStepDetails(
+  replayClient: ReplayClientInterface,
+  filteredEvents: RecordingTestMetadataV3.UserActionEvent[],
+  begin: string,
+  end: string,
+  sources: Map<string, Source>
+): Promise<TestEventDetailsEntry[]> {
+  const testResultPoints = filteredEvents.map(e => e.data.timeStampedPoints.beforeStep!);
+
+  const evalResults: RunEvaluationResult[] = [];
+
+  // This is the injected script that Playwright uses to parse selectors and query the DOM.
+  const playwrightInjectedScriptModule: { source: string } = await import("./injectedScriptSource");
+
+  // These arguments and the preload eval string correspond to the Playwright setup logic
+  // in `playwright-core/src/server/dom.ts::injectedScript()`.
+  // We can hard-code several of these values ourselves.
+
+  // Controls injection into window, and some other external pieces
+  const isUnderTest = false;
+  // Playwright can be used from multiple languages
+  const sdkLanguage = "javascript";
+  // data-testid can be customized by the user
+  const testIdAttributeName = "data-testid";
+  // WebKit on Windows needs 5, everything else is 1
+  const rafCountForStablePosition = 1;
+  const browserName = "chromium";
+  const customSelectorEngines: string[] = [];
+
+  const preloadExpression = `
+        (() => {
+          const module = {};
+          ${playwrightInjectedScriptModule.source}
+          return new (module.exports.InjectedScript())(
+            globalThis,
+            ${isUnderTest},
+            "${sdkLanguage}",
+            "${testIdAttributeName}",
+            ${rafCountForStablePosition},
+            "${browserName}",
+            [${customSelectorEngines.join(",\n")}]
+          );
+        })();
+      `;
+
+  // Any focus range errors here will bubble up to the parent focus cache,
+  // which will retry the load if needed later.
+  await Promise.all(
+    // TODO We don't currently have a way for `runEvaluation` expressions to have variable behavior
+    // per execution point. The old analysis API put `point` and `time` in scope, and if we had
+    // that we could do _one_ `runEvaluation` with a single expression that would contain a lookup
+    // table mapping execution points to locator strings.
+    // Since we don't have that, for now we'll run a separate `runEvaluation` _per point_. This is horrible
+    // for perf, but it at least will work. (Slowly.)
```

Comment:
I think we should maybe hold off on this change for this reason alone.

Reply:
I have a feature request in to the BAC team to expose the current execution point as a variable in scope in evaluations. The old analysis API did that, and `runEvaluation` doesn't. That would let me do one `runEvaluation`, with an injected lookup table of locator strings per execution point.

We _could_ potentially look at using a routine here, although there was something I thought of while working on this that made me think we'd have issues converting this into a routine - can't remember that atm.

Reply:
I'm also concerned about the perf implications because we're bypassing the schedulers. 

If we wanted to land something safe we'd likely need to do it on hover with a denounce to limit the calls.